src/fsort.c

```diff
@@ -252,12 +262,8 @@ SEXP fsort(SEXP x, SEXP verboseArg) {
 
     if (verbose) {
       Rprintf(_("Top 20 MSB counts: ")); for(int i=0; i<MIN(MSBsize,20); i++) Rprintf(_("%"PRId64" "), (int64_t)msbCounts[order[i]]); Rprintf(_("\n"));
-      Rprintf(_("Reduced MSBsize from %zu to "), MSBsize);
-    }
-    while (MSBsize>0 && msbCounts[order[MSBsize-1]] < 2) MSBsize--;
-    if (verbose) {
-      Rprintf(_("%zu by excluding 0 and 1 counts\n"), MSBsize);
     }
+    MSBsize = shrinkMSB(MSBsize, msbCounts, order, verbose);
```

Comment:
I might have assumed this helper would be by-reference on `MSBsize`; OTOH, it's a single `size_t` and this is only done once in `fsort()`. The impact should not be noticed.